update CI versions

lilyminium opened the pull request:
Fixes #146, #147

Changes made in this Pull Request:
 - Updates codecov action version
 - Updates Python versions to 3.11-3.14

I've checked all versions of other actions and they are still up to date as of Jul 2025


PR Checklist
------------
 - [ ] Tests?
 - [ ] Docs?
 - [ ] CHANGELOG.md updated?
 - [ ] AUTHORS.md updated?
 - [x] Issue raised/referenced?

lilyminium committed:
update CI versions

p-j-smith committed:
Merge branch 'main' into fix-issue-147

p-j-smith committed:
Export CODECOV_TOKEN environment variable to pass to CodeCov

p-j-smith reviewed (changes requested):
I think the codecov token is not being set correctly in the `run-cookie-ci` action. I think the `inputs.CODECOV_TOKEN` needs to be set as an environment variable, and the variable passed to `act` rather than the input

p-j-smith committed:
Also pass the GITHUB_TOKEN to act

p-j-smith committed:
Pass GITHUB_TOKEN into run-cookie-ci

orbeckst committed:
updated tested Python in workflow template

- bump to Python 3.14
- comment out the exclusion for 3.14/latest, given that core MDA is fully supporting 3.14

orbeckst reviewed (approved):
The CI is now passing.

I updated Python to 3.14.

I'll merge.

orbeckst commented:
Thank you @lilyminium and @p-j-smith !
